### PR TITLE
Draft change attempting to persist Data Collection Fragment position

### PR DESCRIPTION
**Pull request by JSunde**
Towards #1449

### CODE REVIEW

**Comment by JSunde** [comment]
I think I got the SavedStateHandle set up properly to persist the LiveData storing the current position in the Data Collection Flow, but for some reason when navigating back after capturing a photo for the Photo Task, the DataCollectionViewPagerAdapter is still being passed 0, causing the DataCollectionFragment to navigate back to the first task in the flow.

Any ideas @shobhitagarwal1612, @gino-m?

**Comment by JSunde** [comment]
> I think I got the SavedStateHandle set up properly to persist the LiveData storing the current position in the Data Collection Flow, but for some reason when navigating back after capturing a photo for the Photo Task, the DataCollectionViewPagerAdapter is still being passed 0, causing the DataCollectionFragment to navigate back to the first task in the flow.
> 
> Any ideas @shobhitagarwal1612, @gino-m?

I ended up figuring out the issue. It seems that the DataCollectionViewPagerAdapter was being recreated on resume due to the submission observer. I was able to resolve the issue by only creating a new DataCollectionViewPagerAdapter if there was no adapter set or if the tasklist was changed.

**Comment by JSunde** [comment]
Ready for review, PTAL @gino-m, @shobhitagarwal1612

File: ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionFragment.kt [hunk 1]
```diff
@@ -39,15 +40,20 @@ class DataCollectionFragment : AbstractFragment(), BackPressListener {
...
+    dataCollectionViewModelFactory.create(savedStateHandle)
+  }
+
   private lateinit var viewPager: ViewPager2
 
-  override fun onCreate(savedInstanceState: Bundle?) {
-    super.onCreate(savedInstanceState)
-    viewModel = getViewModel(DataCollectionViewModel::class.java)
-  }
+  //  override fun onResume() {
```
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionFragment.kt:52]: Nit: Remove commented out code.

File: ground/src/test/java/com/google/android/ground/ui/datacollection/DataCollectionViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@HiltAndroidTest
+@RunWith(RobolectricTestRunner::class)
+class DataCollectionViewModelTest : BaseHiltTest() {
+  @Inject lateinit var dataCollectionViewModelFactory: DataCollectionViewModel.Factory
+
+  @Test
+  fun savedState_containsCurrentPosition_returnsCorrectCurrentPosition() {
```
Comment (gino-m) [inline ground/src/test/java/com/google/android/ground/ui/datacollection/DataCollectionViewModelTest.kt:33]: Can you add tests for the null case and for the case when the task list changes?

**Review by gino-m — CHANGES_REQUESTED** [review]

File: ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionFragment.kt (hunk 1 above)
Comment (JSunde) [reply ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionFragment.kt:52]: Oops 😅 

Removed!

File: ground/src/test/java/com/google/android/ground/ui/datacollection/DataCollectionViewModelTest.kt (hunk 2 above)
Comment (JSunde) [reply ground/src/test/java/com/google/android/ground/ui/datacollection/DataCollectionViewModelTest.kt:33]: Hmm... do you have suggestions for how to emit a new value for the submission in the test to trigger the DataCollectionViewPager recreation?

Comment (gino-m) [reply ground/src/test/java/com/google/android/ground/ui/datacollection/DataCollectionViewModelTest.kt:33]: Ah nvm, that happens in the fragment, not the vm. We should eventually add instrumented tests to cover that case.

**Review by gino-m — APPROVED** [review]
